comp/otelcol/ddflareextension/impl/extension.go

```diff
@@ -182,17 +198,37 @@ func (ext *ddExtension) Shutdown(ctx context.Context) error {
 
 // ServeHTTP the request handler for the extension.
 func (ext *ddExtension) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
-	customer, err := ext.configStore.getProvidedConfAsString()
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		fmt.Fprintf(w, "Unable to get provided config\n")
-		return
+	var (
+		customer string
+		err      error
+	)
+	if ext.configStore.isProvidedConfigSupported() {
+		customer, err = ext.configStore.getProvidedConfAsString()
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "Unable to get provided config\n")
+			return
+		}
+	} else {
+		customer = "ConfigProvider not supported, only enhanced config available"
 	}
-	enhanced, err := ext.configStore.getEnhancedConfAsString()
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		fmt.Fprintf(w, "Unable to get enhanced config\n")
-		return
+	var enhanced string
+	if ext.ocb {
+		enhanced, err = ext.configStore.getEffectiveConfigAsString()
+		if err != nil {
+			ext.telemetry.Logger.Error("Unable to get effective config", zap.Error(err))
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "Unable to get effective config\n")
+			return
+		}
+	} else {
+		enhanced, err = ext.configStore.getEnhancedConfAsString()
+		if err != nil {
+			ext.telemetry.Logger.Error("Unable to get enhanced config", zap.Error(err))
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "Unable to get enhanced config\n")
+			return
+		}
```

Comment:
this will be collapsed a little bit since EffectiveConfig will actually go away once mackenzie's changes from [`a90ab38` (#30061)](https://github.com/DataDog/datadog-agent/pull/30061/commits/a90ab38e60479d78c14e6ae43cc70a419f47aeed) are merged in

Reply:
this is removed in favor of using Confmap in the configstore